MVP_Clicker/Assets/Project/Scripts/Game/Areas/Monster/View/MonsterView.cs

```diff
@@ -12,14 +15,30 @@ public class MonsterView : MonoBehaviour, IMonsterView
         [SerializeField] private TextMeshProUGUI _currentHp;
         [SerializeField] private Image _monsterImage;
 
+        public List<Sprite> PullOfMonsterSprites { get; } = new();
+        private IMonsterConfig _config;
+
         public void SetCurrentHp(int currentHp)
         {
             _currentHp.text = Convert.ToString(currentHp);
         }
 
         public void SetMonsterImage(Sprite monsterImage)
         {
-            _monsterImage.sprite = monsterImage;
+            if (monsterImage != null)
+            {
+                _monsterImage.sprite = monsterImage;
+            }
+
+            SetRandomMonsterImage();
```

Comment:
this method works not like expected... If there is image it will not set the image from parameter. The programmer that will use this method does not know about that strange behaviour.

The method must be clear for everyone